nodes/src/nodes/router_llm/router.py

```
@@ -0,0 +1,298 @@
+            # Medium complexity -> tier 2 (balanced)
+            info = _get_model_info(self.primary_model) if _get_model_info(self.primary_model)['tier'] == 2 else _get_model_info('claude-sonnet')
+            info['reason'] = f'complexity score {score}, routed to balanced model'
+        else:
+            # Simple query -> tier 3 (fast/cheap)
+            info = _get_model_info('claude-haiku')
+            info['reason'] = f'complexity score {score} < {self.complexity_threshold // 2}, routed to fast model'
+
+        info['complexity_score'] = score
+        return info
+
+    def _route_cost_aware(self, text: str) -> Dict[str, Any]:
```

Ruff (0.15.7)

[warning] Unused method argument: `text`

(ARG002)